kingpin/actors/aws/iam/entities.py

```diff
@@ -525,23 +580,96 @@ def __init__(self, *args, **kwargs):
         self.get_entity_policy = self.iam_conn.get_user_policy
         self.put_entity_policy = self.iam_conn.put_user_policy
 
+        # Parse the supplied inline policies
+        self._parse_inline_policies(self.option('inline_policies'))
+
+    @gen.coroutine
+    def _ensure_groups(self, name, groups, purge):
+        """Ensure that this user is a member of specific groups.
+
+        args:
+            name: The user we're managing
+            groups: The list (or single) of groups to join be members of
+            purge: Whether or not to purge unmanaged targets.
+        """
+        if isinstance(groups, basestring):
+            groups = [groups]
+
+        current_groups = []
+        try:
+            raw = yield self.thread(self.iam_conn.get_groups_for_user, name)
+            resp_key = 'list_groups_for_user_response'
```

Comment:
All lines that do not raise the exception should not be inside of the try. Either put them into the `else` part of try/catch or just put them outside of the entire block